Utility/Framework/SymfonyFramework.php

```diff
@@ -75,19 +78,6 @@ public static function isKernelLessThan($major, $minor = null, $patch = null)
      */
     private static function kernelVersionCompare($operator, $major, $minor = null, $patch = null)
     {
-        $vernum = $major;
-        $kernel = Kernel::MAJOR_VERSION;
-
-        if ($minor) {
-            $vernum .= '.'.$minor;
-            $kernel .= '.'.Kernel::MINOR_VERSION;
-
-            if ($patch) {
-                $vernum .= '.'.$patch;
-                $kernel .= '.'.Kernel::RELEASE_VERSION;
-            }
-        }
-
-        return version_compare($kernel, $vernum, $operator);
+        return version_compare(Kernel::VERSION_ID, sprintf("%d%'.02d%'.02d", $major, $minor ?: 0, $patch ?: 0), $operator);
```

Comment:
As for simple numeric comparison, that isn't possible due to the nature of this method signature: it allows passing in the comparison operator, hence the usage of `version_compare`.

The `VERSION` constant isn't a bad choice, but it is sometimes followed by a string value like `-dev` and `-beta` and while I know `version_compare` can deal with those, it just doesn't seem as clear to me. From my 100% unscientific polling of other projects, usage of `VERSION_ID` seems to be the most often used way to version detect, which was another reason I'd chosen it originally. 

This change actually originated from a bug in `symfony/symfony#master` where the kernel's `VERSION_MAJOR` was accidentally set to `0` causing our tests to fail: I settled on `VERION_ID` after looking at what others used.

Do you feel that using `VERSION` instead of `VERSION_ID` would add any value?